File: backend/model/Paciente.js
class Paciente {
    constructor(alergias, altura, cpf, data_nascimento, email, foto_perfil = '', nome, peso, sexo,
        circunferencia_abdominal, gordura_corporal, massa_muscular, preferencias, senha) {
        this.alergias = alergias;
        this.altura = altura;
        this.cpf = cpf;
        this.dt_nascimento = data_nascimento;
        this.email = email;
        this.foto_perfil = foto_perfil;
        this.nome = nome;
        this.peso = peso;
        this.sexo = sexo;
        this.circunferencia_abdominal = circunferencia_abdominal;
        this.gordura_corporal = gordura_corporal;
        this.massa_muscular = massa_muscular;
        this.preferencias = preferencias;
        this.senha = senha;

    }

    toFirestore() {
        return {
            alergias: this.alergias,
            altura: this.altura,
            cpf: this.cpf,
            dt_nascimento: this.dt_nascimento,
            email: this.email,
            foto_perfil: this.foto_perfil,
            nome: this.nome,
            peso: this.peso,
            sexo: this.sexo,
            circunferencia_abdominal: this.circunferencia_abdominal,
            gordura_corporal: this.gordura_corporal,
            massa_muscular: this.massa_muscular,
            preferencias: this.preferencias,
            senha: this.senha
        };
    }

    fromJson(json) {
        this.alergias = json.alergias || '';
        this.altura = json.altura || 0;
        this.cpf = json.cpf || '';
        this.dt_nascimento = json.dt_nascimento || '';
        this.email = json.email || '';
        this.foto_perfil = json.foto_perfil || '';
        this.nome = json.nome || '';
        this.peso = parseFloat(json.peso) || 0;
        this.sexo = json.sexo || '';
        this.circunferencia_abdominal = parseFloat(json.circunferencia_abdominal) || 0;
        this.gordura_corporal = parseFloat(json.gordura_corporal) || 0;
        this.massa_muscular = parseFloat(json.massa_muscular) || 0;
        this.preferencias = json.preferencias || '';
        this.senha = json.senha || '';
    }
  }

module.exports = Paciente; // Exporta a classe para uso em outros módulos

File: healthway_app/lib/models/paciente.dart
class Paciente {
  final String? id;
  final String nome;
  final String email;
  final String cpf;
  final String dataNascimento;
  final String sexo;
  final double altura;
  final double peso;
  final double circunferenciaAbdominal;
  final double gorduraCorporal;
  final double massaMuscular;
  final String alergias;
  final String preferencias;
  final String senha;

  Paciente({
    this.id,
    required this.nome,
    required this.email,
    required this.cpf,
    required this.dataNascimento,
    required this.sexo,
    required this.altura,
    required this.peso,
    required this.circunferenciaAbdominal,
    required this.gorduraCorporal,
    required this.massaMuscular,
    required this.alergias,
    required this.preferencias,
    required this.senha,
  });

  factory Paciente.fromJson(Map<String, dynamic> json) {
    return Paciente(
      id: json['id'],
      nome: json['nome'],
      email: json['email'],
      cpf: json['cpf'],
      dataNascimento: json['dataNascimento'],
      sexo: json['sexo'],
      altura: json['altura'].toDouble(),
      peso: json['peso'].toDouble(),
      circunferenciaAbdominal: json['circunferenciaAbdominal'].toDouble(),
      gorduraCorporal: json['gorduraCorporal'].toDouble(),
      massaMuscular: json['massaMuscular'].toDouble(),
      alergias: json['alergias'],
      preferencias: json['preferencias'],
      senha: json['senha'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nome': nome,
      'email': email,
      'cpf': cpf,
      'dataNascimento': dataNascimento,
      'sexo': sexo,
      'altura': altura,
      'peso': peso,
      'circunferenciaAbdominal': circunferenciaAbdominal,
      'gorduraCorporal': gorduraCorporal,
      'massaMuscular': massaMuscular,
      'alergias': alergias,
      'preferencias': preferencias,
    };
  }

  Paciente copyWith({
    String? id,
    String? nome,
    String? email,
    String? cpf,
    String? dataNascimento,
    String? sexo,
    double? altura,
    double? peso,
    double? circunferenciaAbdominal,
    double? gorduraCorporal,
    double? massaMuscular,
    String? alergias,
    String? preferencias,
    String? senha,
  }) {
    return Paciente(
      id: id ?? this.id,
      nome: nome ?? this.nome,
      email: email ?? this.email,
      cpf: cpf ?? this.cpf,
      dataNascimento: dataNascimento ?? this.dataNascimento,
      sexo: sexo ?? this.sexo,
      altura: altura ?? this.altura,
      peso: peso ?? this.peso,
      circunferenciaAbdominal:
          circunferenciaAbdominal ?? this.circunferenciaAbdominal,
      gorduraCorporal: gorduraCorporal ?? this.gorduraCorporal,
      massaMuscular: massaMuscular ?? this.massaMuscular,
      alergias: alergias ?? this.alergias,
      preferencias: preferencias ?? this.preferencias,
      senha: senha ?? this.senha,
    );
  }

  @override
  String toString() {
    return 'Paciente(id: $id, nome: $nome, email: $email, cpf: $cpf, dataNascimento: $dataNascimento, sexo: $sexo, altura: $altura, peso: $peso, circunferenciaAbdominal: $circunferenciaAbdominal, gorduraCorporal: $gorduraCorporal, massaMuscular: $massaMuscular, alergias: $alergias, preferencias: $preferencias)';
  }
}
